src/services/locationiq/api.js

```diff
@@ -0,0 +1,97 @@
+const KEY = process.env.REACT_APP_LOCATION_IQ_KEY;
+const ENDPOINTS = {
+	forwardGeocode: 'https://us1.locationiq.com/v1/search.php?key=',
+  reverseGeocode: 'https://us1.locationiq.com/v1/reverse.php?key='
+};
+
+const encodeAddress = (endpoint, {address, zip, city, state}) => {
+	/*
+  Formats address into a valid query string to append to the endpoint URL.
+  Args: endpoint (str) - url including the api key
+        address components (obj) - object containing address components formatted: {address: '123 fake st.', zip: '94560', city: 'San Jose', state: 'CA'}
+  Return: formatted address query (str)
+  */
+  // need to escape special characters in the address to be a valid URL string
+  const encodedAddress = encodeURIComponent(`${address}, ${city}, ${state} ${zip}`);
+
+  return `${endpoint}&q=${encodedAddress}&format=json`
+}
+
+const encodeLatLng = (endpoint, {lat, lng}) => {
+	/*
+  Formats latitude/longitude coordinates into a valid query string to append to the endpoint URL.
+  Args: endpoint (str) - url including the api key
+        latitude/longitude coordinates (obj) - object containing the latitude and longitude values: {lat: 37.12312, lng: -121.12314}
+  Return: formatted lat/lng query (str)
+  */
+  return `${endpoint}&lat=${lat}&lon=${lng}&format=json`
+}
+
+const responseToJSON = (response) => {
```

Comment:
this might be overkill, we should be able to use the api.js instead of needing to use a combination of await and promises

Reply:
Do you mean using the makeRequest function in src/api/api.js? 

I guess I just wrote it all this way to try and make it a kinda portable encapsulated module.